scripts/speaker_tasks/scp_to_manifest.py

```diff
@@ -30,81 +33,177 @@
-    logging.info("wrote", name)
+    print("wrote", name)
+
+
+def read_file(scp_file, id=-1):
+    json_lines = []
+    with open(scp_file, 'r') as fo:
+        lines = fo.readlines()
+        lines = sorted(lines)
+        for line in lines:
+            line = line.strip()
+            speaker = line.split('/')[id]
+            speaker = list(speaker)
+            speaker = ''.join(speaker)
+            meta = {"audio_filepath": line, "offset": 0, "duration": None, "label": speaker}
+            json_lines.append(meta)
+    return json_lines
+
+
+def read_manifest(manifest):
+    data = []
+    with open(manifest, 'r', encoding='utf-8') as f:
+        for line in f:
+            item = json.loads(line)
+            data.append(item)
+    return data
+
+
+def get_duration(json_line):
+    dur = json_line['duration']
+    if dur is not None:
+        wav_path = json_line['audio_filepath']
+        json_line['duration'] = sox.file_info.duration(wav_path)
+    return json_line
+
+
+def get_labels(lines):
+    labels = []
+    for line in lines:
+        label = line['label']
+        labels.append(label)
+    return labels
+
+
+def write_audio_file(line):
+    filename = line['audio_filepath']
+    label = line['label']
+    offset = line['offset']
+    duration = line['duration']
+    basename = os.path.basename(filename).replace('.wav', '')
+    to_path = os.path.join(CWD, 'chunks', label)
+    os.makedirs(to_path, exist_ok=True)
+    to_path = os.path.join(to_path, basename)
+    final_string = '_' + str(offset) + '_' + str(duration)
+    final_string = final_string.replace('.', '-')
+    samples, sr = sf.read(filename)
+    start = int(float(offset * sr))
+    end = start + int(float(duration * sr))
+    chunk = samples[start:end]
+    to_file = to_path + final_string + '.wav'
+    sf.write(to_file, chunk, sr)
+
+    line['offset'] = 0
+    line['audio_filepath'] = to_file
+    return line
 
 
-def main(scp, id, out, split=False, create_chunks=False):
+def main(scp, manifest, id, out, split=False, create_chunks=False, write_chunks=False, min_count=10, workers=4):
     if os.path.exists(out):
         os.remove(out)
-    scp_file = open(scp, 'r').readlines()
-    scp_file = sorted(scp_file)
-
-    lines = []
-    speakers = []
-    for line in tqdm(scp_file):
-        line = line.strip()
-        y, sr = l.load(line, sr=None)
-        dur = l.get_duration(y=y, sr=sr)
-        speaker = line.split('/')[id]
-        speaker = list(speaker)
-        speaker = ''.join(speaker)
-        meta = [{"audio_filepath": line, "offset": 0, "duration": float(dur), "label": speaker}]
-        speaker = [speaker]
-        if create_chunks:
-            meta, speaker = filter_manifest_line(meta[0], signal=y, sr=sr)
-        lines.extend(meta)
-        speakers.extend(speaker)
+    if scp:
+        lines = read_file(scp_file=scp, id=id)
+    else:
+        lines = read_manifest(manifest)
+
+    # lines = process_map(get_duration, lines, chunksize=100)
```

Comment:
This seems like you mistakenly left not uncommented.

Reply:
Good catch, it should have been updated while I was experimenting